crates/verify/src/bytecode.rs

```diff
@@ -567,7 +567,7 @@ fn try_match(
     has_metadata: bool,
 ) -> Result<(bool, Option<VerificationType>)> {
     // 1. Try full match
-    if *match_type == VerificationType::Full && local_bytecode.starts_with(bytecode) {
+    if *match_type == VerificationType::Full && local_bytecode == bytecode {
```

Comment:
We should also add a test to prevent regression in the future, since existing tests did not catch this